modules/kafka-cluster/main.tf

```diff
@@ -1,25 +1,36 @@
-locals {
-  build_cluster_type_basic    = var.cluster_for_production ? [] : ["yes"]
-  build_cluster_type_standard = var.cluster_for_production ? ["yes"] : []
-}
-
 resource "confluent_kafka_cluster" "cluster" {
   display_name = var.kafka_cluster_name
   availability = var.availability
   cloud        = var.cloud
   region       = var.region
 
   dynamic "basic" {
-    for_each = local.build_cluster_type_basic
+    for_each = var.cluster_type == "basic" ? [1] : []
     content {}
   }
 
   dynamic "standard" {
-    for_each = local.build_cluster_type_standard
+    for_each = var.cluster_type == "standard" ? [1] : []
     content {}
   }
 
+  dynamic "dedicated" {
+    for_each = var.cluster_type == "dedicated" ? [1] : []
```

Comment:
Minor: I think these three strings (`"dedicated"`, `"standard"`, `"basic"`) could be made into constants and sit together one after the other. But minor, you might not want to code it that way.

Reply:
the reason being is that it will come from workspace env that is the case. We want to pass the value depending upon the cluster creation type.

Reply:
@bibek4699 It's just a pattern we follow across our terraform repos.  Instead of having "magic" strings in multiple places that must match, we use locals as a kind of `CONST` to make sure there are no typos anywhere.  In this case we would use it both here and in the validations.

Reply:
@mxfive I have done the changes